Pidgin.Tests/ParserTestBase.cs

```diff
@@ -18,5 +19,18 @@ public class ParserTestBase
             Assert.Equal(expectedError, result.Error);
             Assert.Equal(consumedInput, result.ConsumedInput);
         }
+
+        protected void AssertSuccess<TToken, T>(Result<TToken, T> result, string expected, bool consumedInput)
+        {
+            Assert.True(result.Success);
+            Assert.Equal(expected, result.Value.ToString());
+            Assert.Equal(consumedInput, result.ConsumedInput);
+        }
+
+        protected void AssertFailure<TToken, T>(Result<TToken, T> result, bool consumedInput)
+        {
+            Assert.False(result.Success);
+            Assert.Equal(consumedInput, result.ConsumedInput);
+        }
```

Comment:
What was wrong with the existing AssertSuccess/AssertFailure methods?

Reply:
My test expressions are complex, and would have required so much more care in defining the expected expression tree as to defeat the purpose of the test being reliable, I also wanted to quickly add new tests. But this is something that I can move as internal methods of the test class, there is no need to use the base class you have written.